src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,179 @@
+# This file is part of RLinearAlgebra.jl
+
+# Using the structs, MAInfo, in the file that implementing MA. 
+include("solve_log_ma.jl")
+
+"""
+    LSLogFullMA <: LinSysSolverLog
+
+A mutable structure that stores information about a randomized linear solver's behavior.
+    The log assumes that the full linear system is available for processing. The goal of
+    this log is usually for research, development or testing as it is unlikely that the
+    entire residual vector is readily available.
+
+# Fields
+- `collection_rate::Int64`, the frequency with which to record information to append to the
+    remaining fields, starting with the initialization (i.e., iteration 0).
+- `resid_hist::Vector{Float64}`, retains a vector of numbers corresponding to the residual
+    (uses the whole system to compute the residual)
+- `resid_norm::Function`, a function that accepts a single vector argument and returns a
+    scalar. Used to compute the residual size.
+- `iterations::Int64`, the number of iterations of the solver.
+- `converged::Bool`, a flag to indicate whether the system has converged by some measure
+- `ma_info::MAInfo`, [`MAInfo`](@ref)
+- `lambda_hist::Vector{Int64}`, contains the widths of the moving average.
+- `iota_hist::Vector{Float64}`, contains an estimate used for calculating the variability
+    of the progress estimators. These values are stored at iterates specified by
+    `collection_rate`.
```

Comment:
This is not necessary since the residuals are not random conditioned on the iterates.
Check with @nathanielpritchard